guides/jupyter-notebooks.md

```diff
@@ -50,137 +117,305 @@ notebooks/
 ## Section 1
 
-Intro to section 1
+Intro to Section 1
 
-### Sub-section 1a
+### Subsection 1a
 
 More detailed info about Section 1
 
 ## Section 2
 
-A complete thought that's as important as section 1 but doesn't need sub-sections.
+A complete thought that's as important as Section 1 but doesn't need
+subsections.
+
 ```
 
-Be sure to use section headings (i.e., the number of `#`'s) in a way that gives hierarchical meaning to your document.  The header levels are used to do things like intelligently make links, so you don't want to confuse them by using heder levels that don't match the logical flow of the document.
+Be sure to use the Markdown headings (that is, the number of `#`'s) in a way
+that gives hierarchical meaning to your tutorial. The header levels are used to
+do things like intelligently make links and the Table of Contents, so you don't
+want to confuse things by using header levels that don't match the logical flow
+of the tutorial.
 
-### Exercises
+All terms and relevant additional resources should be defined and linked to as
+new topics are introduced and your tutorial progresses, so that terms and
+resources appear within the context of your main content. Short exercises can be
+woven into your Main Content sections as well.
 
-Most notebooks are trying to convey *some* information to their reader.  It's often a good idea, if possible, to include in the notebook something where there is *not* an included answer so that the reader can cement in their mind how whatever it is they were supposed to learn from that notebook. If you do have exercise(s), be sure to leave a blank code cell underneath to show the user that it's meant for them to try it out right there.  You may also want to include a "solutions" notebook next to your main notebook for the user to look at after they have finished their attempt.
+### Exercises
 
-### Additional resources
+Exercises are optional, but encouraged. Exercises can be woven into the Main
+Content of your tutorial, or appear in their own section toward the end of the
+tutorial. Final exercises can be more challenging, similar to homework problems.
+They can be minimal or take as long as 30 minutes to an hour to complete.
+
+Most tutorials convey information to their reader by posing questions and then
+providing answers. But as so many of us learn by doing, it's encouraged for you
+to provide exercises in your tutorial that pose questions but do not include
+immediate answers so that your reader can practice their new skills to cement
+the knowledge in their minds.
+
+If you do have one or more exercises, be sure to leave a blank code cell
+underneath each to show the reader that they're meant to try out their new skill
+right there. You may also want to include a "solutions" notebook next to your
+main notebook for the reader to check their work after they have finished their
+attempt.
+
+### Additional Resources
+
+.. suggestion to remove the Additional Resources section and instead ask
```

Comment:
I agree with this broad point, but I think it may make sense to leave this in as a catch-all for things that really don't fit cleanly into the main narrative.  Perhaps specifically highlight everywhere "additional resources" is mentioned (including the template) that it's "optional", and that it's only for things you can't fit elsewhere?